src/lib/pure_pursuit/PurePursuit.hpp

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2024 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <matrix/math.hpp>
+
+using namespace matrix;
+
+/**
+ * @file PurePursuit.hpp
+ *
+ * Implementation of pure pursuit guidance logic.
```

Comment:
Could you add a reference to the formulation you are using? There are variants of the pure pursuit, and it would be nice if this is already clarified in the comments.

Reply:
This library simply returns the heading to the intersection point between a circle around the vehicle and the line segment between the previous and next waypoint.
Correct me if I am wrong, but the differences between the pure pursuit variants are how this heading is translated into a steering input and how the lookahead radius adapts to i.e. different vehicle speeds. 
This is covered in the individual modules (since it is different for each) and not in the library.

Reply:
I know, but if you can cite a paper that is consistent to the formulation you are using, it would be helpful for people understanding the implementation.

Reply:
added in [ec4923f](https://github.com/PX4/PX4-Autopilot/pull/23387/commits/ec4923f129fe29d3e21d639c281a2c3b36511420)